crates/wallet/src/wallet/mod.rs

```
@@ -356,25 +395,64 @@ impl Wallet {
+    ///
+    /// # Errors
     ///
-    /// Refer to [`Wallet::create`] for more.
-    pub fn create_with_params(params: CreateParams) -> Result<Self, DescriptorError> {
+    /// This returns a [`DescriptorError`] in a number of situations:
+    ///
+    /// - If parsing a [`Descriptor`] fails
+    /// - If there's a mismatch in the length of two or more multipath specifiers
+    /// - If a multipath descriptor appears where one was not expected
+    /// - If the descriptor and change descriptor are the same
+    ///
+    /// [multipath]: miniscript::Descriptor::is_multipath
+    /// [`keymap`]: CreateParams::keymap
+    /// [`change_descriptor`]: CreateParams::change_descriptor
+    /// [`MultiXPub`]: miniscript::DescriptorPublicKey::MultiXPub
+    /// [`Descriptor`]: miniscript::Descriptor<DescriptorPublicKey>
+    pub fn create_with_params(mut params: CreateParams) -> Result<Self, DescriptorError> {
         let secp = SecpCtx::new();
         let network = params.network;
         let genesis_hash = params
             .genesis_hash
             .unwrap_or(genesis_block(network).block_hash());
         let (chain, chain_changeset) = LocalChain::from_genesis_hash(genesis_hash);
 
-        let (descriptor, mut descriptor_keymap) = (params.descriptor)(&secp, network)?;
+        let (mut descriptor, mut descriptor_keymap) = (params.descriptor)(&secp, network)?;
+
+        // In case of multipath we parse the descriptor into single descriptors and
+        // assign each one back to the `descriptor` and `change_descriptor`.
+        if descriptor.is_multipath() {
+            let mut desc_iter = descriptor.into_single_descriptors()?.into_iter();
+            descriptor = desc_iter.next().expect("must have a descriptor");
+            descriptor_keymap = Descriptor::parse_descriptor(&secp, &descriptor.to_string())?.1;
+            params.change_descriptor = desc_iter.next().map(params::make_descriptor_to_extract);
+        }
```

I think it should throw an error if there are more than 2 descriptors.

---

Although you've explained that only the first 2 are used in the documentation, I also wonder if returning an error would be a better idea in order to prevent the user's unexpected behavior.

---

Can you make a code suggestion (preferably non breaking)?

---

I can't say I fully agree with throwing an error, and instead lean toward managing expectations through effective documentation. But I guess it can go either way.